packages/dashboard-core-plugins/src/panels/NotebookPanel.tsx

```diff
@@ -681,23 +681,40 @@ class NotebookPanel extends Component<NotebookPanelProps, NotebookPanelState> {
     this.focus();
   }
 
-  handleCopy(): void {
+  async handleCopy(): Promise<void> {
+    const { fileStorage, glEventHub, session } = this.props;
     const { fileMetadata, settings } = this.state;
     assertNotNull(fileMetadata);
-    const content = this.getNotebookValue();
     const { language } = settings;
     const { itemName } = fileMetadata;
-    const copyName = FileUtils.getCopyFileName(itemName);
+    let copyName = FileUtils.getCopyFileName(itemName);
+    // await in loop is fine here, this isn't a parallel task
+    // eslint-disable-next-line no-await-in-loop, @typescript-eslint/strict-boolean-expressions
+    while (await FileUtils.fileExists(fileStorage, copyName)) {
+      copyName = FileUtils.getCopyFileName(copyName);
+    }
     log.debug('handleCopy', fileMetadata, itemName, copyName);
-    this.createNotebook(copyName, language, content);
+    await fileStorage.copyFile(itemName, copyName);
```

Comment:
Ideally we'd be catching an error if one occurred here, and displaying a toast for it. However, we don't really have toasts wired up at all at the moment, and it's possible we just wire up a toast service for catching uncaught exceptions anyway so do not worry about it for now.